commit pure translations separately

SimonHeimberg opened the pull request:
**Type of change**
Enhancement

**Issue(s) closed**
replaces #6040 

**To reproduce**
Try to review the translations in 0c2abcc5b6. Quite difficult because there are many line changes as well. 

**New behavior**
Edits of translations are committed separately from updating the files from the code.
This allows to review the translations much simpler.

**How it works**
*For features and functional changes:*
- Fetch the translations from transifex
- commit those if there are any (with a identifiable commit message)
- update the translation template files (.pot)
- push them to transifex
- fetch the updated translations
- (update other files like statistics and authors)
- commit translations files and their templates (with changes related to each other)

**Possible regressions**
- a bit more noise because the same translation file might be committed twice in a row
- no other known

**Additional context**
not sure about the commit messages: `Fetched translations.` and `Updated translations catalogs.`
(before `Fetched translations and updated catalogs.`)

I included some clean-up as well (related to #6063) because having edits and cleanup of one file in separate pr results in merge conflicts.

SimonHeimberg committed:
improve merge_and_push_translations.sh with shellcheck

- do not use "if [ $? -eq 0 ]; then" since this does not work with "set -e". SC2181
- quote variable. SC2086
- Use ! grep -q instead of comparing its output with [ -z .. ]. SC2143

SimonHeimberg committed:
merge_and_push_translations.sh pulls only newer translations from transifex

This works after a fresh checkout by using --use-git-timestamps,
the timestamps of the files are not helpful then.

SimonHeimberg committed:
pure translations and source changes are committed separately

This makes it easier to review translations.

SimonHeimberg committed:
cleanup, replace tabs with spaces in merge_and_push_translations.sh

SimonHeimberg committed:
merge_and_push_translations.sh only pushes if any commit was created

SimonHeimberg committed:
merge_and_push_translations.sh works in other locale than English

tothxa reviewed (commented):
I think I like the idea, but I'd feel safer if we got other opinions whether the double commits are OK.

SimonHeimberg committed:
merge_and_push_translations.sh always does tx force pull

to really make sure that all translations are received

and only pull them the 2nd time after uploading anything

SimonHeimberg committed:
merge_and_push_translations.sh undoes changes only when there could be

and not anyway

SimonHeimberg committed:
merge_and_push_translations.sh does not ignore failure when there should be none

SimonHeimberg committed:
merge_and_push_translations.sh with update_ and git add calls in functions

preparation for the next commit

SimonHeimberg committed:
merge_and_push_translations.sh lets git expand the paths

The command line can not get too long.
The output looks nicer because of the shorter command line.

To have each file listed which is added, run "git add -v FILES"

SimonHeimberg committed:
merge_and_push_translations.sh generates files on matching changes

Files (authors, appdata, statistics) is generated for a commit of
translation and for a commit of catalogs.

Desired effects:
When translations are edited, only commit "translation" is created.
When translation sources change, only commit "catalog" is created.
When both change, both commits are created.
When a file was changed (on release, ...) and no translation changes,
  the commit "catalog" is created.
No commit triggers the other one.

SimonHeimberg committed:
revert loop in merge_and_push_translations.sh

Revert two statements changed in d8870109.
This can be discussed and improved in a separate pull request.

SimonHeimberg committed:
show variable functionality clearer in merge_and_push_translations.sh

- put it close to the statements it belongs to
- check git status to see if the update scripts need to run a 2nd time
  (hint by tothxa)

and style:
- use snake_case (not camelCase), as for other variables in the script

SimonHeimberg committed:
consistently run undo_oneliner_diffs in merge_and_push_translations.sh

run it after both calls of tx pull

suggested by tothxa

SimonHeimberg committed:
more obvious check for commits in merge_and_push_translations.sh

suggested by tothxa

SimonHeimberg committed:
initialize variable in utils/merge_and_push_translations.sh

It is good style.

Co-authored-by: Tóth András <80712255+tothxa@users.noreply.github.com>

SimonHeimberg committed:
Apply suggestions from code review to merge_and_push_translations.sh

- changes from git go into the "translation" commit
- all update_* scripts are run before a potential "catalog" commit

Co-authored-by: Tóth András <80712255+tothxa@users.noreply.github.com>

SimonHeimberg committed:
more comments in merge_and_push_translations.sh

and remove an echo whose message does not match anymore

tothxa committed:
clean-up: only run updates in catalogs turn if anything changed

tothxa committed:
give feedback when nothing is done
